Analysis/bin/CreateSpectralHists.cxx

```diff
@@ -41,21 +44,167 @@ struct HistArgs {
   }
 };
 
-class HistSpectrum : public root_ext::AnalyzerData {
 
-public:
-  HistSpectrum(std::shared_ptr<TFile> outputFile, const HistArgs& hargs) : root_ext::AnalyzerData(outputFile)
-  {
-    eta_pt_hist.SetMasterHist(hargs.eta_bins, hargs.eta_min, hargs.eta_max,
-                              hargs.pt_bins, hargs.pt_min, hargs.pt_max);
+namespace analysis {
+
+struct HistSpectrum : public root_ext::AnalyzerData {
+  using Tau = tau_tuple::Tau;
+  using TauTuple = tau_tuple::TauTuple;
+
+  HistSpectrum(std::shared_ptr<TFile>& outputFile, const HistArgs& hargs) : root_ext::AnalyzerData(outputFile) {
+    eta_pt_hist.SetMasterHist(
+      hargs.eta_bins, hargs.eta_min, hargs.eta_max,
+      hargs.pt_bins, hargs.pt_min, hargs.pt_max);
+  };
+
+  virtual ~HistSpectrum() {};
```

Comment:
```suggestion
```

Reply:
About this I am a bit confused, why it gives me a warning like `has virtual functions and accessible non-virtual destructor`, in my understanding it is to prevent forgetting of calling the destructor for the `back to base class cast` like `Base* b = new Derived();` but you mean that in our case we are safe because we're only defining Derived class, right?

Reply:
If the base class has a virtual destructor, there should be no need to define empty virtual destructor in child classes as it must be created by default by the compiler. And `AnalyzerData` [has one](https://github.com/cms-tau-pog/TauMLTools/blob/master/Core/interface/AnalyzerData.h#L85). I'm puzzled why would you are getting this warning... it is still there if you define `HistSpectrum` as `class`?

Reply:
My bad, I am saying in general, for instance for my SourceDesc<T> class, but now I understood, thank you.

Reply:
Yes, in general, if class has virtual functions, you should also define a virtual destructor. But doing it once in the base class is sufficient.